blob/service.go

```diff
@@ -186,6 +205,49 @@ func (s *Service) Included(
 	return true, resProof.equal(*proof)
 }
 
+// Subscribe returns all blobs under the given namespaces at subscrubed heigh.
+// Subscribe can return map of blobs and an error in case if some requests failed.
+func (s *Service) Subscribe(ctx context.Context, namespaces []share.Namespace) (<-chan map[uint64]BlobsByNamespace, error) {
+	headerChan, err := s.headerSubscribe(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	blobChan := make(chan map[uint64]BlobsByNamespace)
+	go func() {
+		defer close(blobChan)
+		for {
+			select {
+			case head := <-headerChan:
+				res := make(map[uint64]BlobsByNamespace)
+
+				wg := sync.WaitGroup{}
+				for i, namespace := range namespaces {
+					wg.Add(1)
+					go func(i int, namespace share.Namespace) {
+						defer wg.Done()
+						blobs, err := s.getBlobs(ctx, namespace, head)
+						if err != nil {
+							log.Debugw("error getting blobs", "namespace", namespace.String(), "height", head.Height())
+							return
+						}
```

Comment:
In case of an error, the returned value will lack some data, making it impossible for the subscriber to identify whether there were errors or data missed in this scenario.

One way to handle this situation could be to shut down the subscription and return the error on the first occurrence. While this is a common approach, it would mean that the subscriber, upon encountering the first error, would need to resubscribe and could potentially miss some critical blocks during the process. I would suggest handling these errors differently.

I agree with the point @distractedm1nd raised about returning a single struct that contains both the blob and height. Perhaps the returned struct should also include information about errors. This would allow the subscription to continue even in the presence of errors, shifting the error-handling logic to the subscriber. The subscriber would still have the ability to shut down the subscription if the handling logic deems it necessary.

Reply:
@walldiss I added extra channel for errors, including height, namespace and error itself. Since subscriber will decide himself to unsubscribe channel or not.